File: book_review_web/frontend/src/services/authorService.js
import api from './api';

export const authorService = {
  getAuthors: async (page = 0, size = 10) => {
    const response = await api.get('/authors', { params: { page, size } });
    return response.data;
  },
  
  createAuthor: async (data) => {
    const response = await api.post('/authors', data);
    return response.data;
  },
  
  updateAuthor: async (id, data) => {
    const response = await api.put(`/authors/${id}`, data);
    return response.data;
  },
  
  deleteAuthor: async (id) => {
    const response = await api.delete(`/authors/${id}`);
    return response.data;
  }
};

File: book_review_web/backend/src/main/java/com/bookreview/service/AuthorService.java
package com.bookreview.service;

import com.bookreview.dto.AuthorDTO;
import com.bookreview.dto.PageResponseDTO;

public interface AuthorService {
    PageResponseDTO<AuthorDTO> getAllAuthors(int page, int size);
    AuthorDTO createAuthor(AuthorDTO authorDTO);
    AuthorDTO updateAuthor(Long id, AuthorDTO authorDTO);
    void deleteAuthor(Long id);
}
